zeebe/gateway-grpc/src/main/java/io/camunda/zeebe/gateway/interceptors/impl/AuthenticationInterceptor.java

```diff
@@ -45,14 +56,96 @@ public <ReqT, RespT> Listener<ReqT> interceptCall(
     final var authorization = headers.get(AUTH_KEY);
     if (authorization == null) {
       LOGGER.debug(
-          "Denying call {} as no token was provided", methodDescriptor.getFullMethodName());
+          "Denying call {} as no authentication information were provided",
+          methodDescriptor.getFullMethodName());
       return deny(
           call,
           Status.UNAUTHENTICATED.augmentDescription(
-              "Expected bearer token at header with key [%s], but found nothing"
+              "Expected authentication information at header with key [%s], but found nothing"
                   .formatted(AUTH_KEY.name())));
     }
+    // the authentication can be of 2 types, oidc or basic.
+    if (isBasicAuth(authorization)) {
+      return handleBasicAuth(call, headers, next, methodDescriptor, authorization);
+    }
+    return handleOidcAuth(call, headers, next, methodDescriptor, authorization);
+  }
+
+  private boolean isBasicAuth(final String authorization) {
+    return authorization.startsWith("Basic ");
+  }
+
+  private <ReqT, RespT> Listener<ReqT> handleBasicAuth(
+      final ServerCall<ReqT, RespT> call,
+      final Metadata headers,
+      final ServerCallHandler<ReqT, RespT> next,
+      final MethodDescriptor<ReqT, RespT> methodDescriptor,
+      final String authorization) {
+    try {
+      final String basicAuth = authorization.replaceFirst("Basic ", "");
+      final var decodedAuth = new String(Base64.getDecoder().decode(basicAuth));
+      final String[] authParts = decodedAuth.split(":", 2);
+      final String username = authParts[0];
+      final String password = authParts[1];
+
+      final var userOpt = loadUserByUsername(username);
+      if (userOpt.isEmpty()) {
+        LOGGER.debug(
+            "Denying call {} as the user {} does not exist",
+            methodDescriptor.getFullMethodName(),
+            username);
+        return deny(
+            call,
+            Status.UNAUTHENTICATED
+                .augmentDescription("Expected a valid user, but user does not exist")
+                .withCause(new IllegalArgumentException("User does not exist")));
+      }
+
+      final var user = userOpt.get();
+      if (!isPasswordValid(password, user.password())) {
+        LOGGER.debug(
+            "Denying call {} as the password is not valid for user {}",
+            methodDescriptor.getFullMethodName(),
+            user.username());
+        return deny(
+            call,
+            Status.UNAUTHENTICATED
+                .augmentDescription("Expected a valid password, but password is not valid")
+                .withCause(new IllegalArgumentException("Password is not valid")));
```

Comment:
❌ From a security view you want to provide a potential hacker with a little information as possible, whilst still making it clear to the user what the problem is. By saying here that the password is not valid, you are implicitly telling someone that the username they are working with is correct. It'd be better to change this error message to somethng like:

> Invalid credentials